### ISSUE TITLE
Poly Haven causes Blender to crash

### ISSUE THREAD

**Issue by hokutonanto** [Poly-Haven/polyhavenassets#81]
**Describe the bug**
If all PolyHaven assets are present, Blender crashes when trying to access the asset browser.

**Console log**

[blender.crash.txt](https://github.com/Poly-Haven/polyhavenassets/files/12383543/blender.crash.txt)

**Screenshots**
<img width="863" alt="image" src="https://github.com/Poly-Haven/polyhavenassets/assets/23059969/337be19e-c1ed-4ad6-96d8-8b212e2c8741">

**Additional context**
This crash only happens when Poly Haven assets are active.
After lots of attempts at understanding the issue, and a few Blender bug tickets submitted, I've been able to screen capture the console before Blender quits.

It's clearly due to something broken in the Poly Have assets, but I'm not sure how to fix the issue.

In the screen capture I attached, you can see what happens the moment I try to access the asset browser.

https://github.com/Poly-Haven/polyhavenassets/assets/23059969/a4f2802f-6387-4fd2-b2c8-406d60ed3658

**Comment by gregzaal** [comment]
Hi there,

I can't reproduce this problem, it loads fine on my machine (also Win 11 in Blender 3.6.2).

It's possible the file in the error message (`coast_land_rocks_02.blend`) is corrupted somehow, could you try to delete that asset folder and click "Fetch Assets" again?

**Comment by hokutonanto** [comment]
Thanks, @gregzaal 

I deleted that asset, but the issue still remains.

It worked fine only while I had a few assets downloaded. Once I decided to download all of them, that the crash appeared.

**Comment by gregzaal** [comment]
Your error message was specifically related to `coast_land_rocks_02.blend`, if it's crashing again now, what is the new error message? 

Assuming it is another similarly corrupted file, you can try running this in a Python console in Blender:

```py
bpy.ops.pha.pull_from_polyhaven(asset_type="models", revalidate=True)
```

![image](https://github.com/Poly-Haven/polyhavenassets/assets/5604661/6295c4ac-4433-4aaf-9d37-1a752fd2d1c1)

Revalidation will check the hashes of all downloaded files and redownload them if needed.

**Comment by hokutonanto** [comment]
We're getting something, this is what happened.
When I downloaded the first batch, and for some reason, it stopped at some point, the assets downloaded were good.
When I finished downloading the rest, I noticed the connection wasn't as smooth as the first time — and no matter how many times I try it always drops.

So, yes, there are many corrupted assets.

I then run the script you suggested, but I'm getting this error:

`Error: Python: Traceback (most recent call last):
  File "\Text", line 1, in <module>
NameError: name 'bpy' is not defined`

**Comment by hokutonanto** [comment]
Ops... you're right, I was using the TextEditor.

I run it through the console, and it seems it downloaded the assets with no problem this time.
Apparently, it's not crashing anymore. I'll test it a few more times.

What changed with the python command you gave me compared to the other way the Poly Haven plugin downloads the assets?

**Comment by jessey-git** [comment]
Blender received 2 bugs on this recently, 1 from another Poly Haven user, and should now be handled a bit better (no crash) in 3.6.3 when that releases next month.

It would manifest if the file load encountered a "Missing DNA block" (a legitimate error that the file is indeed corrupted) and would then incorrectly crash inside `is_minversion_older_than_blender` -- the crash should now be fixed at least.

Bug: https://projects.blender.org/blender/blender/issues/111235
Bug: https://projects.blender.org/blender/blender/issues/111246
Fix: https://projects.blender.org/blender/blender/commit/d9831a9a4bb1723ab3e2d17d819400c8301a913f

**Comment by hokutonanto** [comment]
@jessey-git glad to hear this will be handled better in 3.6.3

@gregzaal I noticed a new asset today, and I tried to download it. Something broke again, so I tried your python command one more time, only, now it gets stuck here:

`Blender quit
Read blend: "E:\Assets\B3D\Assets\Poly Haven\round_wooden_table_01\round_wooden_table_01.blend"
WARN: Could not make abandoned_factory_canteen_01_4k.exr relative to E:\Assets\B3D\Assets\Poly Haven\round_wooden_table_01
WARN: Could not make wooden_lounge_1k.hdr relative to E:\Assets\B3D\Assets\Poly Haven\round_wooden_table_01
Info: Saved "round_wooden_table_01.blend"`

I opened the **round_wooden_table_01.blend** and the file loads fine, except for the HDRI **abandoned_factory_canteen_01_4k.exr** that is missing. The path is off (I see the typical purple missing texture).

Should I simply re-link the HDRI and save the file, or is that going to mess up the path even more?

**Comment by gregzaal** [comment]
Fixed in `v1.1.5` - various checks were added to avoid and retry corrupted downloads, as well as a "Revalidate All Assets" in the user preferences in case this can't be accessed in the Asset Browser due to crashing.

Blender 3.6.3 should also no longer crash outright if there are corrupted assets in the browser.

@hokutonanto Those are only normal warnings, not errors, and should not affect the usage of the add-on. If you have another issue using the add-on, please open a separate bug report.

**Comment by hokutonanto** [comment]
Hm... it crashes right when it tries to access that asset:
https://github.com/Poly-Haven/polyhavenassets/assets/23059969/101dd697-160f-4673-a254-e546ffe64eea

[blender.crash.txt](https://github.com/Poly-Haven/polyhavenassets/files/12410542/blender.crash.txt)


https://github.com/Poly-Haven/polyhavenassets/assets/23059969/88cd1f29-ae00-4547-aed0-6d35350bf325
